### DISCUSSION TITLE
Model server is not responding quickly because of some threading issues

### DISCUSSION THREAD

**Discussion by shinesamjacob** [eclipse-emfcloud/emfcloud#216] [ModelServer] [answered]
I am getting the following error while closing the glsp editor, any clue like why these errors are printed

![image](https://github.com/eclipse-emfcloud/emfcloud/assets/19471062/1cdc5665-7d8b-4138-839f-013c47b9b5a3)

**Comment by CamilleLetavernier** [comment, marked as answer]
This is a very generic error that happens when an action takes time to execute (As the label suggests :) ). It may indicate a deadlock (if the message keeps getting printed forever), or just a very slow operation. In your case, it doesn't seem to loop forever, so it just looks like a slow operation.

The only way to figure it out is to debug the model server when this happens, to know which operation is slow and why.
